hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/mapred/TestCopyCommitter.java

```diff
@@ -160,10 +160,10 @@ public void testPreserveStatus() throws IOException {
       context.setTargetPathExists(false);
 
       CopyListing listing = new GlobbedCopyListing(conf, CREDENTIALS);
-      Path listingFile = new Path("/tmp1/" + String.valueOf(rand.nextLong()));
+      Path listingFile = new Path("/tmp1/" + rand.nextLong());
       listing.buildListing(listingFile, context);
 
-      conf.set(DistCpConstants.CONF_LABEL_TARGET_WORK_PATH, targetBase);
+      conf.set(DistCpConstants.CONF_LABEL_TARGET_FINAL_PATH, targetBase);
```

Comment:
Why we changed the path here ?? Assuming this test is for non-atomic, values of CONF_LABEL_TARGET_FINAL_PATH and CONF_LABEL_TARGET_WORK_PATH will be same.

Reply:
This test case is not the case of Atomic Copy or non Atomic Copy. You are right about that, the value of final Path & work path would be same in case of non atomic copy . But as you see above we are setting the environment for testing the preserve status functionality & not doing a copy. It solely depends on the configuration value of the final Path & not the work path. This is the bug I raised & fixed the test case pertaining to this along with it.

Reply:
Thanks got it. Code was wrong as well as the testcase was wrong. Thanks for fixing.

Reply:
@mukund-thakur  Can this code be merged? Or is there anything I have to do from my end.